valhalla/app/controllers/concerns/valhalla/resource_controller.rb

```diff
@@ -52,8 +52,10 @@ def after_delete_success
     end
 
     def edit
-      @change_set = change_set_class.new(find_resource(params[:id])).prepopulate!
+      @change_set = change_set_class.new(find_resource(params[:id]))
       authorize! :update, @change_set.resource
+      @change_set.validate({})
+      @change_set.prepopulate!
```

Comment:
So prepopulate should happen right after creating the change_set (per slack discussion). Maybe this will have implications for the changeset property definitions? I'm not sure.